pkg/runner/test_runner_route.go

```diff
@@ -58,7 +50,8 @@ func (r *stepRouter) routeIn(terminate <-chan struct{}) (int, error) {
 
 	// terminateTargetWriter is a control channel used to signal termination to
 	// the writer object which injects a target into the test step
-	terminateTargetWriter := make(chan struct{})
+	terminateTargetWriterCtx, terminateTargetWriter := context.WithCancel(context.Background())
+	defer terminateTargetWriter() // avoids possible goroutine deadlock in context.WithCancel implementation
```

Comment:
Why do we need this, can this ever fire before we hit the `terminateTargetWriter()` below?

Reply:
This change is not required, I just decided to get rid off channels.

This can't fire before we reach terminateTargetWriter() because we create a channel around context.Background() (not on the input ctx) which never gets cancelled.

So, it should be the same logic as it was before, but just using context.

Reply:
What I meant was, why do we need `defer terminateTargetWriter()` here, when a couple lines later we unconditionally call the same function anyway (terminateTargetWriter()). This is before the defer would fire.